addons/website_slides/models/slide_slide.py

```diff
@@ -35,6 +35,13 @@ class SlidePartnerRelation(models.Model):
     completed = fields.Boolean('Completed')
     quiz_attempts_count = fields.Integer('Quiz attempts count', default=0)
 
+    @api.constrains('vote')
+    def _check_vote(self):
+        """ The value of the vote must be 1, 0 and -1 otherwise raise an error """
+        for slide_partner in self:
+            if slide_partner.vote not in (1, 0, -1):
+                raise ValidationError(_('The vote must be 1, 0 or -1. %d is not allowed value for vote', slide_partner.vote))
+
```

Comment:
It should be done with `_sql_constraints` I think (it's more a technical constrain and it's more performant in SQL)

```
_sql_constraints = [
    ('check_vote', 'CHECK(vote IN (-1, 0, 1))', 'The vote must be 1, 0 or -1.'),
]
```

(I'm not sure if you should do a migration script to make each negative value to -1 and each positive value to 1)